activesupport/lib/active_support/ordered_options.rb

```diff
@@ -92,18 +92,48 @@ def inspect
   #   h.boy  # => 'John'
   class InheritableOptions < OrderedOptions
     def initialize(parent = nil)
-      if parent.kind_of?(OrderedOptions)
+      @parent = parent
+      if @parent.kind_of?(OrderedOptions)
         # use the faster _get when dealing with OrderedOptions
-        super() { |h, k| parent._get(k) }
-      elsif parent
-        super() { |h, k| parent[k] }
+        super() { |h, k| @parent._get(k) }
+      elsif @parent
+        super() { |h, k| @parent[k] }
       else
         super()
+        @parent = {}
       end
     end
 
+    def to_h
+      @parent.dup.merge(self)
+    end
+
+    def ==(other)
+      to_h == other.to_h
+    end
+
+    def inspect
+      "#<#{self.class.name} #{to_h.inspect}>"
+    end
+
+    def key?(key)
+      super || @parent.key?(key)
+    end
+
+    def overridden?(key)
+      !!(@parent && @parent.key?(key) && keys.include?(key.to_sym))
```

Comment:
nitpick:
```suggestion
      !!(@parent&.key?(key) && keys.include?(key.to_sym))
```

Also, calling `keys.include?` is a bit much.

We could do something like:

```ruby
alias_method :own_key?, :key?
private :own_key?

def key?(key)
  # ...
end

def overridden?(key)
  !!(@parent&.key?(key) && own_key?(key.to_sym))
end
```

Reply:
Oh `own_key?` is much better.